src/components/structures/RoomView.js

```diff
@@ -143,17 +150,46 @@ module.exports = React.createClass({
             }
         });
 
+        if (this.props.roomAddress[0] == '#') {
+            // we always look up the alias from the directory server:
```

Comment:
No - we still join by the address that was passed in (ie. the alias in this case), we just need to look up the ID for things like peeking.